homeassistant/components/waterfurnace.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for Waterfurnace component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/waterfurnace/
+"""
+from datetime import timedelta
+import logging
+import time
+import threading
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+
+REQUIREMENTS = ["waterfurnace==0.1.0"]
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "waterfurnace"
+UPDATE_TOPIC = DOMAIN + "_update"
+CONF_UNIT = "unit"
+SCAN_INTERVAL = timedelta(seconds=10)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_UNIT): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, base_config):
+    """Setup waterfurnace platform."""
+    import waterfurnace.waterfurnace as wf
+    config = base_config.get(DOMAIN)
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    unit = config.get(CONF_UNIT)
+
+    wfconn = wf.WaterFurnace(username, password, unit)
+    wfconn.login()
```

Comment:
What happens if login fails?

Reply:
Right now, a KeyError is thrown because sessionid is not found in the cookies. It's not elegant, but it will stop the rest of the execution.